docs/src/appendix/migrating-from-chiseltest.md

```diff
@@ -0,0 +1,98 @@
+Unfortunately, Chisel 3's testing library, [ChiselTest](https://github.com/ucb-bar/chiseltest), is built around the SFC, making it difficult to support ChiselTest in Chisel 5 and beyond.
+
+This is not a fundamental problem, it is one of engineering resources.
+This page is intended to serve as a living document to help Chisel users migrate from ChiselTest to Chisel's native simulation library, ChiselSim.
+
+## Getting Started
+
+The developers of ChiselTest have maintained some amount of compatibility between ChiselTest and newer versions of Chisel.
+This relies on a forked version of the old SFC so there may be compatibility issues, but it does provide some versions of feature overlap for users to migrate their code.
+
+We recommend using the latest minor version of Chisel 5 and ChiselTest 5 for migrating. At the time of writing this is Chisel v5.1.0 and ChiselTest v5.0.2, but please check for later versions.
+
+## Migration
+
+ChiselSim provides a minimal `peek`, `poke`, `expect`, and `step` API, similar to that of ChiselTest.
+You use ChiselSim by importing it `import chisel3.simulator.EphemeralSimulator._`, and using its `simulate` method which is similar to ChiselTest's `test`.
+At present, it does not have any integration with ScalaTest, so users should use any ScalaTest APIs directly.
+
+For example, given a simple design (typically in `src/main/scala`):
+
+```scala mdoc:silent
+import chisel3._
+class MyModule extends Module {
+  val io = IO(new Bundle {
+    val in = Input(UInt(16.W))
+    val out = Output(UInt(16.W))
+  })
+
+  io.out := RegNext(io.in)
+}
+```
+
+The legacy ChiselTest way to test this would be with a `ChiselScalatestTester` in `src/test/scala`:
+
+<!-- This cannot be mdoc because we do not compile against chiseltest -->
+```scala
+import chisel3._
+import chiseltest._
+import org.scalatest.flatspec.AnyFlatSpec
+
+class MyModuleSpec extends AnyFlatSpec with ChiselScalatestTester {
+  behavior of "MyModule"
+  it should "do something" in {
+    test(new MyModule) { c =>
+      c.io.in.poke(0.U)
+      c.clock.step()
+      c.io.out.expect(0.U)
+      c.io.in.poke(42.U)
+      c.clock.step()
+      c.io.out.expect(42.U)
+      println("Last output value : " + c.io.out.peek().litValue)
+    }
+  }
+}
+```
+
+This can be rewritten using ChiselSim as follows:
+```scala mdoc:silent
+import chisel3._
+import chisel3.simulator.EphemeralSimulator._
+import org.scalatest.flatspec.AnyFlatSpec
+
+class MyModuleSpec extends AnyFlatSpec {
+  behavior of "MyModule"
+  it should "do something" in {
+    simulate(new MyModule) { c =>
+      c.io.in.poke(0.U)
+      c.clock.step()
+      c.io.out.expect(0.U)
+      c.io.in.poke(42.U)
+      c.clock.step()
+      c.io.out.expect(42.U)
+      println("Last output value : " + c.io.out.peek().litValue)
+    }
+  }
+}
+```
```

Comment:
It is great that the APIs here are so similar. 👍  This goes a long way towards showing that some migrations may be relatively painless.